### PR TITLE
some console reporting improvements

### PR DESCRIPTION
**Pull request by jquense**
Trying to consolidate the reporting code into a single place. Ultimately we need to get a handle on which errors should crash the process and which shouldn't, but first we need them to all go through one place

### CODE REVIEW

**Comment by jquense** [comment]
image

pretty errors!

**Comment by KyleAMathews** [comment]
Hallelujah!

This looks great.

BTW, while you're working on this — for any GraphQL-only error we should eliminate the stack trace since they're meaningless.

**Comment by jquense** [comment]
image

image

File: packages/gatsby/src/utils/webpack.config.js [hunk 1]
```diff
@@ -166,6 +166,7 @@ module.exports = async (
           // ids to reduce filesize.
           new webpack.NamedModulesPlugin(),
           new FriendlyErrorsWebpackPlugin({
+            clearConsole: false,
```
Comment (jquense) [inline packages/gatsby/src/utils/webpack.config.js]: I turned this off because the webpack output was clearing gatsby errors very quickly

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 470f3a246d647366893a061ebba2e373a8884174

https://app.netlify.com/sites/using-glamor/deploys/598b71267960b13d69f4bf3a

**Comment by jquense** [comment]
I think this is good? I don't know if the deploy errors are me, Appveyor failure is a snapshot test thing due to another PR

**Comment by jquense** [comment]
might reorganize the code a bit...

File: packages/gatsby/src/utils/webpack.config.js (hunk 1 above)
Comment (KyleAMathews) [inline packages/gatsby/src/utils/webpack.config.js]: 💯 yeah I've heard/helped several people that got stuck because they never saw the errors

**Comment by KyleAMathews** [comment]
Woah, there's a ton of awesome stuff in here!! Looking forward to landing this! And consolidating all our reporting will make it way easier to iterate on this stuff moving forward.

**Comment by jquense** [comment]
I think this is good? Not sure about the appveyor case

**Comment by KyleAMathews** [comment]
I broke appveyor and just fixed it in https://github.com/gatsbyjs/gatsby/pull/1752 so we're good here. Merging!!!

**Pull request merged**